docs/source/examples/word-count.rst

```diff
@@ -0,0 +1,219 @@
+Word count
+==========
+
+In this example, we'll use ``dask`` to count the number of words in text files
+(Enron email dataset, 6.4 GB) both locally and on a cluster with HDFS (along
+with the `distributed`_ and `hdfs3`_ libraries).
+
+Local computation
+-----------------
+
+Download the first text file (76 MB) in the data set to your local machine:
+
+.. code-block:: bash
+
+   $ wget https://s3.amazonaws.com/blaze-data/enron-email/edrm-enron-v2_allen-p_xml.zip/merged.txt
+
+Import ``dask.bag`` and create a ``bag`` from the single text file:
+
+.. code-block:: python
+
+   >>> import dask.bag as db
+   >>> b = db.from_filenames('merged.txt')
+
+View the first ten lines of the text file with ``take``:
+
+.. code-block:: python
+
+   >>> b.take(10)
+
+   ('Date: Tue, 26 Sep 2000 09:26:00 -0700 (PDT)\r\n',
+    'From: Phillip K Allen\r\n',
+    'To: pallen70@hotmail.com\r\n',
+    'Subject: Investment Structure\r\n',
+    'X-SDOC: 948896\r\n',
+    'X-ZLID: zl-edrm-enron-v2-allen-p-1713.eml\r\n',
+    '\r\n',
+    '---------------------- Forwarded by Phillip K Allen/HOU/ECT on 09/26/2000 \r\n',
+    '04:26 PM ---------------------------\r\n',
+    '\r\n')
+
+Generate a word count expression using the ``bag`` methods to split the lines
+into words, concatenate the nested lists of words into a single list, count the
+frequencies of each word, then list the top 10 words by their count:
+
+.. code-block:: python
+
+   >>> wordcount = b.str.split().concat().frequencies().topk(10, lambda x: x[1])
+
+Note that the combined operations in the previous expression are lazy. We can
+trigger the word count computation using ``.compute()``:
+
+.. code-block:: python
+
+   >>> wordcount.compute()
+
+   [('P', 288093),
+    ('1999', 280917),
+    ('2000', 277093),
+    ('FO', 255844),
+    ('AC', 254962),
+    ('1', 240458),
+    ('0', 233198),
+    ('2', 224739),
+    ('O', 223927),
+    ('3', 221407)]
+
+Cluster computation with HDFS
+-----------------------------
+
+Next, we'll use ``dask`` along with the `distributed`_ with the `hdfs3`_
+libraries to count the number of words in all of the text files stored in HDFS.
+
+Copy the text data from S3 into HDFS on the cluster:
+
+.. code-block:: bash
+
+   $ hadoop distcp s3n://{AWS_SECRET_ID}:{AWS_SECRET_KEY}@blaze-data/enron-email hdfs:///tmp/enron
+
+where ``AWS_SECRET_ID`` and ``AWS_SECRET_KEY`` are valid AWS credentials.
+
+Start the ``distributed`` scheduler and workers on the cluster. Because of the
+parallelizable nature of ``bag`` operations, we can start the ``distributed``
+workers with multiple processes equal to the number of cores on the machine:
```

Comment:
"Because our computations use Pure Python rather than numeric libraries (numpy, pandas, etc.) we start workers with multiple processes rather than with multiple threads.  This helps us avoid issues with the Python Global Interpreter Lock and increases efficiency."